support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -89,12 +80,19 @@ object CreateZuoraSubscription {
     impure: Impure
   ): Future[HandlerResult[SendThankYouEmailState]] = {
     import impure._
-    val corporateCodeValidator = CorporateCodeValidator.withDynamoLookup(redemptionService)
+    val subscriptionDataBuilder = {
+      val corporateCodeValidator = CorporateCodeValidator.withDynamoLookup(redemptionService)
+      val dsPurchaseBuilder = new DigitalSubscriptionPurchaseBuilder(config.digitalPack, promotionService, giftCodeGenerator, today)
+      val dsRedemptionBuilder = new DigitalSubscriptionCorporateRedemptionBuilder(corporateCodeValidator, today)
+      new SubscriptionBuilder(dsPurchaseBuilder, dsRedemptionBuilder, promotionService, config.contributionConfig _)
+    }
```

Comment:
I did try some arbitrary groupings in case classes like I did in the SendThankYouEmail equivalent, but none of them were really clear.
I think splitting between per-request and reusable parameters makes most sense, as then the business decision making logic is not obscured by promoServices and configs.
This also means we are more likely to stay within the 8 parameter limit and 50 lines limit.